db/pebble/pebble.go

```diff
@@ -102,15 +106,17 @@ func prefixIterOptions(prefix []byte) *pebble.IterOptions {
 }
 
 // Iterate implements the method Iterate of the interface db.Storage
-func (p *PebbleStorage) Iterate(f func([]byte, []byte) (bool, error)) error {
+func (p *Storage) Iterate(f func([]byte, []byte) (bool, error)) (err error) {
 	// NewIter already provides a point-in-time view of the current DB
 	// state, but if is used for long term (is not the case), should use an
 	// iterator over an snapshot:
 	// snapshot := p.pdb.NewSnapshot()
 	// defer snapshot.Close()
 	// iter := snapshot.NewIter(nil)
 	iter := p.pdb.NewIter(prefixIterOptions(p.prefix))
-	defer iter.Close()
+	defer func() {
```

Comment:
This is problematic.  If there's an error in this function, we won't get the error but the result of `iter.Close()` which may be nil!  This is replacing the real error.  It should be something like:
```
err1 := iter.Close()
if err != nil {
  return err
} else {
  return err1
}
```

Reply:
Right! Thanks!